packages/snaps-execution-environments/src/common/endowments/interval.ts

```diff
@@ -17,13 +17,15 @@ const createInterval = () => {
         `The interval handler must be a function. Received: ${typeof handler}`,
       );
     }
+    harden(handler);
     const handle = Object.freeze({});
     const platformHandle = setInterval(handler, timeout);
     registeredHandles.set(handle, platformHandle);
-    return handle;
+    return harden(handle);
```

Comment:
this makes freezing above redundant.

on the other hand, it's better to harden right away, so you should replace handle line with one of:
```js
    const handle = Object.freeze(Object.create(null));
```
or
```js
  const handle = harden({});
```

The first option being preferred as more explicit

Reply:
I have added now the first option proposed with `const handle = Object.freeze(Object.create(null));`.
It is updated in both, timeout and interval endowments.